pkg/validator/validate.go

```diff
@@ -0,0 +1,189 @@
+package validator
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/ident"
+)
+
+var (
+	ReValidRef          = regexp.MustCompile(`^[^\s]+$`)
+	ReValidBranchID     = regexp.MustCompile(`^\w[-\w]*$`)
+	ReValidRepositoryID = regexp.MustCompile(`^[a-z0-9][a-z0-9-]{2,62}$`)
+)
+
+var (
+	ErrInvalid           = errors.New("validation error")
+	ErrInvalidType       = fmt.Errorf("invalid type: %w", ErrInvalid)
+	ErrRequiredValue     = fmt.Errorf("required value: %w", ErrInvalid)
+	ErrInvalidValue      = fmt.Errorf("invalid value: %w", ErrInvalid)
+	ErrPathRequiredValue = fmt.Errorf("missing path: %w", ErrRequiredValue)
```

Comment:
@nopcoder I thought about it, the problem with this approach is that it might mask some meaningful errors, for instance, the ValidatePath function returns 3 different errors (ErrInvalidType, ErrPathRequiredValue and a custom one wrapping the ErrInvalidValue) and I can't really handle it properly from the caller function as it fails fast (first validation error returns).
I think we can do some work here and rethink of error handling (there are some packages that handle it better compared to the built-in one).
If you think we can still achieve custom errors without loosing visibility, I'd appreciate to hear more on your thoughts

Reply:
I moved the package-specific validations into the packages scope, this way, the errors, types, etc would be managed by the package.
I left the common resources (regex, general functions, etc) in the validate package